File: 198-house-robber/198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        int n = nums.length;
        if(n == 1) return nums[0];
        int pprev = nums[0], prev = Math.max(nums[0], nums[1]);
        
        for(int i = 2; i < n; ++i) {
            int current = Math.max(nums[i] + pprev, prev);
            pprev = prev;
            prev = current;
        }
        
        return prev;
    }
}

File: 198-house-robber/198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        prev2, prev = 0, nums[0]
        for idx in range (1, n):
            current = max(nums[idx] + prev2, prev)
            prev2 = prev
            prev = current
        return prev